zrml/swaps/src/lib.rs

```diff
@@ -1360,6 +1362,14 @@ mod pallet {
             let mut map = BTreeMap::new();
             let mut total_weight = 0;
             let amount_unwrapped = amount.unwrap_or_else(BalanceOf::<T>::zero);
+            // Sort assets for future binary search, for example to check if an asset is included.
+            let sort_assets = assets.as_mut_slice();
+            sort_assets.sort();
+            let has_duplicates = sort_assets
+                .iter()
+                .zip(sort_assets.iter().skip(1))
+                .fold(false, |acc, (&x, &y)| acc || x == y);
+            ensure!(!has_duplicates, Error::<T>::SomeIdenticalAssets);
```

Comment:
Sorry, just noticed another problem. You need to either clone `assets` or put this code blob _after_ the next `if` expression, because the order of `assets` needs to remain unchanged until the weight map is created in case the pool uses CPMM as scoring rule.

For design reasons, I would make `assets` non-mutable and clone it to prevent these types of errors in the future.

Reply:
good catch 👍